packages/app/src/specs/banners/TrackedBanner.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+  <Alert
+    v-bind="$attrs"
+    :model-value="modelValue"
+    @update:model-value="handleBannerDismissed"
+  >
+    <slot />
+  </Alert>
+</template>
+
+<script setup lang="ts">
+import Alert from '@packages/frontend-shared/src/components/Alert.vue'
+import { watch } from 'vue'
+import { gql, useMutation, useQuery } from '@urql/vue'
+import { TrackedBanner_ProjectStateDocument, TrackedBanner_SetProjectStateDocument } from '../../generated/graphql'
+import { set } from 'lodash'
+
+type AlertComponentProps = InstanceType<typeof Alert>['$props']
+type AlertComponentEmits = InstanceType<typeof Alert>['$emit']
+interface TrackedBannerComponentProps extends AlertComponentProps {
+  bannerId: string
+  modelValue: boolean
+}
+interface TrackedBannerComponentEmits extends AlertComponentEmits {
+  (e: 'update:modelValue'): void
+}
+
+gql`
+query TrackedBanner_ProjectState {
+  currentProject {
+    id
+    savedState
+  }
+}
+`
+
+gql`
+mutation TrackedBanner_SetProjectState($value: String!) {
+  setPreferences(type: project, value: $value) {
+    ...TestingPreferences
+    ...SpecRunner_Preferences
+  }
+}
+`
+
+const props = withDefaults(defineProps<TrackedBannerComponentProps>(), {})
+
+const emit = defineEmits<TrackedBannerComponentEmits>()
+
+const stateQuery = useQuery({ query: TrackedBanner_ProjectStateDocument })
+const setStateMutation = useMutation(TrackedBanner_SetProjectStateDocument)
+
+watch(
```

Comment:
I think we can use `watchEffect`, looks similar to `watch` but `immediate: true` by default. https://vuejs.org/api/reactivity-core.html#watcheffect

Interestingly enough there is also `watchPostEffect` and `watchEffectSync` which I haven't used before. Good to know about, I think it's like React's `flushSync`, [docs](https://reactjs.org/docs/react-dom.html#flushsync).

```js
watchEffect(() => {
  if (props.modelValue) {
    updateBannerState('lastShown')
   }
})
```